docs/analysis/perfetto-sql-getting-started.md

```diff
@@ -2,8 +2,23 @@
 
 PerfettoSQL is the foundation of trace analysis in Perfetto. It is a dialect of
 SQL that allows you to query the contents of your traces as if they were a
-database. This page introduces the core concepts of PerfettoSQL and provides
-guidance on how to write queries.
+database. This page introduces the core concepts of trace querying with PerfettoSQL
+and provides guidance on how to write queries.
+
+## Overview of Trace Querying
+
+The Perfetto UI is a powerful tool for visual analysis, offering call stacks, timeline views, thread tracks, and slices. However, it also includes a robust SQL query engine (PerfettoSQL) that allows you to extract data programmatically.
+
+While the UI is powerful for myriad of analyses, users are able to write and execute queries within the Perfetto UI for multiple purposes such as:
+
+- Extracting performance data from traces.
+- Create custom visualizations (Debug tracks) to perform more complex analyses.
+- Creating derived metrics.
+- Identify performance bottlenecks using data-driven logic.
+
+It is also possible to query programmatically instead of using the Perfetto UI, querying programmatically is currently supported in C++ with the [C++ Trace Processor](trace-processor.md) and via python using the [Python Trace Processor APIs](trace-processor-python.md).
```

Comment:
I would mention Python first (as the preferred interface) and C++ as an advanced usecase